### ISSUE TITLE
Add SaveBang rule to rubocop and correct errors

### ISSUE THREAD

**Issue by martinlovell** [yalelibrary/YUL-DC#2000]
There are places where `save`, `update`, and `destroy` are called without checking the return value.
It's likely those are unintended, and those should probably be converted to the matching `save!`, `update!`, and `destroy!`, or there should it should check the return code.

There is a rubocop rule to check for these issues: `RuboCop::Cop::Rails::SaveBang`

- [x] Add the following rule to `.rubocop.yml`:
```
Rails/SaveBang:
  Enabled: true
  Exclude:
    - 'spec/**/*.rb'
```

- [x] Add rubocop ignore to places where this rule incorrectly finds a `save` in non-active record code
- [x] Fix all the rubocop issues found by this change related to active record

Generally, call `save!`, especially if you want the Error to bubble up.  It's also possible to check the return value of save and check `object.errors` to determine what went wrong.  There are probably some places where we want to rescue the Error to prevent if from bubbling up (or check the return value of save).

(There are 24 offenses detected, in about 14 files.)

Thanks to @jpengst for pointing out this problem.

**Comment by jpengst** [comment]
PR: https://github.com/yalelibrary/yul-dc-management/pull/1078

**Comment by martinlovell** [comment]
Released management v2.59.68
